src/coreclr/runtime/amd64/AllocFast.S

```diff
@@ -234,7 +234,7 @@ LEAF_ENTRY RhpNewPtrArrayFast, _TEXT
 
         // Delegate overflow handling to the generic helper conservatively
 
-        cmp         rsi, (0x40000000 / 8) // sizeof(void*)
+        cmp         rsi, 0x8000000 // (0x40000000 / sizeof(void*))
```

Comment:
We build with both llvm and gcc successfully. What's the compiler used by SunOS that this needs fixing?

Reply:
I tried making a simple, stand-alone example test.S file with the original constant but I don't see an easy way to replicate all the CPP (and other?) pre-processing that's being used in this code.

Reply:
BTW, that line compiles OK if I just remove the parentheses:
```        cmp         rsi, 0x40000000 / 8
```